### ISSUE TITLE
Support saving/loading gstbe output pipelines

### ISSUE THREAD

**Issue by ohel** [quodlibet/quodlibet#3002]
I've been thinking of making a couple of enhancements: first to the preferences where one could name and save/load output pipelines, second to the startup options where one could configure which pipeline preset to use via command line parameter.

There are three use cases that at least I use almost daily, all involve a system with bare ALSA (no PulseAudio):
1. I use a gstreamer parametric equalizer in the pipeline for my room equalization if listening via loudspeakers. I don't use it if I'm using headphones.
2. I use two sound cards, and sometimes I wish to output sound to the other.
3. If I'm using Bluetooth headphones, I must use a specific bluez-alsa device.

I've been using a startup script to achieve this functionality by modifying the config file before launching Quod Libet, so I kind of have a working setup already. Also the parametric equalizer I could implement as a LADSPA plugin and the default ALSA device via environment variables and asoundrc config file, but it feels like a lot of moving parts.

Would changing these from the preferences make any sense to anyone else, or does it involve such a special use case it's not worth it?

**Comment by lazka** [comment]
I don't mind exposing device selection through CLI parameters. But please don't use the pipeline string for this, ideally it should go away as it's just a workaround for missing features.

**Comment by ohel** [comment]
I figure by "missing features" you mean the possibility to select ALSA device graphically in the preferences dialog, e.g. from a list of devices?

That's the usual way of handling the business I guess. However, what I do like about Quod Libet is exactly the possibility to define a custom gstreamer pipeline, not just the output device. When I'm listening via speakers, I use the followind pipeline:
> audiocheblimit mode=high-pass cutoff=20 poles=4 type=1 ripple=0 ! audioconvert ! equalizer-nbands num-bands=3 band1::freq=118 band1::bandwidth=6.0 band1::gain=-5.0 ! alsasink device=hifi

What I would like is to be able to save the whole string as a pipeline preset, and switch the pipeline preset via CLI instead of just the device. But I do understand if this is a special setup and not exactly worth the trouble.

**Comment by lazka** [comment]
> I figure by "missing features" you mean the possibility to select ALSA device graphically in the preferences dialog, e.g. from a list of devices?

Yeah. We could also expose "--list-devices", "--set-device" options for CLI then.

What about having a `QUODLIBET_GSTBE_PIPELINE` env var we read on start for now?

**Comment by ohel** [comment]
I'm currently using the sed stream editor to directly tweak the QL config file before launching QL. Using the env var might be a more user friendly step forward. I wonder how common a scenario tweaking the pipeline actually is for people, or am I the only one.

What about the ALSA device selection, are there any plans for it already? How should it be accomplished? Show the selections from ```aplay -L```? Also, should there be a placeholder string in the pipeline for the selected device, or would the pipeline always implicitly end in "! alsasink device=<your device selection>"? Although, I guess with different platforms and backends things will get really complicated quickly.
